docs/mgmt/readme.md

```diff
@@ -689,6 +710,64 @@ public static partial class ResourcesExtension
+
+By default, because the generator will never find the `VirtualMachineResource` when generating this SDK, the parent resource of this `VmGuestConfigurationResource` will be `ResourceGroupResource`.
+```csharp
+public static partial class GuestConfigurationExtensions
+{
+    public static VmGuestConfigurationCollection GetVmGuestConfigurations(this ResourceGroupResource resourceGroup, string vmName)
+    {
+        /* ... */
+    }
+}
+```
+
+To show the relationship between resources across different RPs, we could convert it into a scope resource by using the following configuration:
+```yaml
+parameterized-scopes:
+- /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}
+```
+This configuration registers the listed request paths as resources the generator could recognize even if they might not exist in the current context. After applying this configuration, the generated code will have the following changes:
+```diff
+public static partial class GuestConfigurationExtensions
+{
+-   public static VmGuestConfigurationCollection GetVmGuestConfigurations(this ResourceGroupResource resourceGroup, string vmName)
+-   {
+-       /* ... */
+-   }
++   public static VmGuestConfigurationCollection GetVmGuestConfigurations(this ArmClient client, ResourceIdentifier scope)
++   {
++       if (!scope.ResourceType.Equals("Microsoft.Compute/virtualMachines"))
++           throw new InvalidOperationException(string.Format("Invalid resource type {0} expected Microsoft.Compute/virtualMachines", scope.ResourceType));
++       /* ... */
++   }
+}
+```
+
+This configuration works fine with the configuration introduced above:
+```yaml
+generate-arm-resource-extensions:
+- /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachines/{vmName}/providers/Microsoft.GuestConfiguration/guestConfigurationAssignments/{guestConfigurationAssignmentName}
+```
+and this is how the generated code would change:
+```diff
+public static partial class GuestConfigurationExtensions
+{
+    public static VmGuestConfigurationCollection GetVmGuestConfigurations(this ArmClient client, ResourceIdentifier scope)
+    {
+        if (!scope.ResourceType.Equals("Microsoft.Compute/virtualMachines"))
+            throw new InvalidOperationException(string.Format("Invalid resource type {0} expected Microsoft.Compute/virtualMachines", scope.ResourceType));
+        /* ... */
+    }
++
++   public static VmGuestConfigurationCollection GetVmGuestConfigurations(this ArmResource armResource)
++   {
++       /* ... */
++   }
```

Comment:
Theorectically, the combination of the two configurations work fine. But the extension method would only work for `VirtualMachineResource`, others would throw exception, right? Then I wonder if there are real cases that need to use both configurations. We'd better not mention it in the doc if no real usages.

Reply:
OK sorry I misunderstood your comment in the first place.
Yes, I planned to add validations before the ArmResource extension method as those scope methods so that they could only call it without exceptions on VirtualMachineResource.
I also planned to add more information in the generated xml document to emphasize that.

Reply:
I mean do we ever need to add an extension method to `ArmResource` if the extension method only works for `VirtualMachineResource`? From functionality perspective, it works. Keeping the case in the tests is also fine. My suggestion is that if we do not have real usage case of ever needing such extension methods, we may omit the case of using both configurations in the doc. Or if there may be real cases, please mention when we need to consider adding the `ArmResource` extension methods in the doc.

Reply:
In this case, no. The configuration just provide the possibility.
In the real case, we do not add this `generate-arm-resource-extensions` configuration unless we could confirm that the scope could be a lot of resources.

Reply:
let me add something in the document to explain that a bit more by telling the reader that despite we could do this, but it is not recommended.